flytekit/image_spec/image_spec.py

```diff
@@ -52,6 +54,11 @@ def image_name(self) -> str:
             container_image = f"{self.registry}/{container_image}"
         return container_image
 
+    def is_container(self) -> bool:
+        if os.environ.get(_F_IMG_ID):
```

Comment:
if will return true in all cases where `_F_IMG_ID` is not set
For example 
```python
@task 
def foo():
  ...

i = ImageSpec(...)

if i.in_container():
   @task(container_image=)
   def bar()
      ...
```

When task `foo` is loaded at runtime, then `bar` will also be loaded right?
should we also add 
```
if not os.environ.get(_F_IMG_ID):
  if current_context().execution_id:
       return False
  return True
else return os.environ.get(_F_IMG_ID) == self.image_name()
```